src/lambda/commands/deploySamApplication.ts

```diff
@@ -30,40 +31,71 @@ export async function deploySamApplication({
     }
 
     const { template, s3Bucket, stackName } = args
+    const deployApplicationPromise = new Promise<void>(async (resolve, reject) => {
 
-    const tempFolder = await mkdtemp('samDeploy')
-    const outputTemplatePath = path.join(tempFolder, 'template.yaml')
-    let stage = 'packaging'
-    try {
-        const packageInvocation = new SamCliPackageInvocation(template.fsPath, outputTemplatePath, s3Bucket, invoker)
-        restParams.outputChannel.show(true)
-        // TODO: Add nls support
-        restParams.outputChannel.appendLine(`Packaging SAM Application to S3 Bucket: ${s3Bucket}`)
-        await packageInvocation.execute()
+        const tempFolder = await mkdtemp('samDeploy')
+        const buildDestination = path.join(tempFolder, 'build')
+        const buildTemplatePath = path.join(buildDestination, 'template.yaml')
+        const outputTemplatePath = path.join(tempFolder, 'template.yaml')
+        let stage = 'starting up'
+        try {
+            restParams.outputChannel.show(true)
+            stage = 'building'
+            const build = new SamCliBuildInvocation(
+                buildDestination,
+                undefined,
```

Comment:
What's the purpose of omitting the build dir? Is this sensitive to the CWD?

Reply:
We omit the base directory.

The template can have multiple resources, and you cannot specify a base directory per resource. At this point you're relying on the template (and all resource CodeUris) to be correctly configured.

From the [docs](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html), "By default, relative paths are resolved with respect to the template's location."